### ISSUE TITLE
Cannot OTA update Philips end-devices

### ISSUE THREAD

**Issue by SHxKM** [Koenkk/zigbee2mqtt#10139]
## What happened
I am completely unable to OTA update _any_ devices like the [Philips motion sensor  - 9290012607](https://www.zigbee2mqtt.io/devices/9290012607.html#philips-9290012607) (currently 4 paired) and the [Hue dimmer switch - 324131092621](https://www.zigbee2mqtt.io/devices/324131092621.html#philips-324131092621) (currently 8 paired) for quite a while now.

## What did you expect to happen

To be able to update these devices via Zigbee2MQTT.

## How to reproduce it (minimal and precise)

1. Pair a device like `9290012607` (firmware I have: `6.1.1.27575`) or `324131092621` (firmware I have: `5.45.1.17846`)
2. Go to OTA screen in Z2M UI
3. Click  "update device firmware"
4. Wait a while while inspecting the logs

Logs, each appear at a different stage:

```
Updating 'buckram_quad_remote' to latest firmware
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Updating 'buckram_quad_remote' to latest firmware","meta":{"device":"buckram_quad_remote","status":"update_in_progress"},"type":"ota_update"}'
```

```
Debug Updating to latest '0x0017880108f1d611' (RWL021)
Debug Using endpoint '2'
```

```
Error Update of 'buckram_quad_remote' failed (Device didn't respond to OTA request)
Debug Error: Device didn't respond to OTA request at requestOTA (/app/node_modules/zigbee-herdsman-converters/lib/ota/common.js:196:15) at Object.updateToLatest (/app/node_modules/zigbee-herdsman-converters/lib/ota/common.js:278:21) at OTAUpdate.onMQTTMessage (/app/lib/extension/otaUpdate.ts:234:21)
```

## Debug info
Zigbee2MQTT version: 1.22.1-1
Adapter hardware: Conbee II

**Comment by Bascht74** [comment]
@SHxKM 
Did you press any button on the dimmers after you click update? 
I had to wake up the devices for OTA...

I don't know if there is anything in Z2M, like -> "after pressing the OTA button wait for the device to wake up by itself (e.g. report battery) and then send an OTA-command". 
For me it usually times out after some time...

Seb

**Comment by SHxKM** [comment]
@Bascht74,

> Did you press any button on the dimmers after you click update?

I have, it rarely helps. Even if it does, the operation will time out after little progress.


> Did you press any button on the dimmers after you click update?

> For me it usually times out after some time...

So does this mean clicking a button works for you, or does not?

**Comment by Bascht74** [comment]
> So does this mean clicking a button works for you, or does not?

Yes, that worked for me for all devices…

**Comment by SHxKM** [comment]
> > So does this mean clicking a button works for you, or does not?
> 
> Yes, that worked for me for all hue devices…

Unfortunately does not work for me. Clicking any button before/after and both before and after the update command does not work.

**Comment by SHxKM** [comment]
bump

**Comment by SHxKM** [comment]
**Update (Dec 28 2021):**

- For motion sensors (9290012607) I can sometimes start the process after trying a few times, but most of the time it won't finish the OTA update.
- For the remotes (324131092621) it's still quite impossible to OTA update.

@Koenkk any idea why this might be happening? is there any way to "force" the device to respond to OTA request?

**Comment by jeroenhoebe** [comment]
After also getting the same Error. I was finally able to update my remote 324131092621 to (6.1.1.28573) through the OTA update. 

It didn't go smoothly though. Needed to press a button every few minutes so it wouldn't lose connection. Then after it finished nothing really changed. But in the logs the status said no update available anymore.

Then I needed to get it to update in zigbee2mqtt because it didn't show the correct firmware yet. So I tried a few things ( remove remote, re-insert battery, holding setup till it flashed, re-pairing) and then after two tries I saw that is was updated. (maybe not exactly in that order, I tried a few things).

**Comment by SHxKM** [comment]
> It didn't go smoothly though. Needed to press a button every few minutes so it wouldn't lose connection.


I have tried this exactly numerous times, without success.

**Comment by Tonio16** [comment]
Hello

For 324131092621 I can update it but though z2m write it is successfull, the firmware version is the same and thedevice is stillavaillble for a new update. Realy strange behavior.
For philipps or ikea bulbs no issue.


Antoine

**Comment by SHxKM** [comment]
@Koenkk - do you recognize these bindings? you told me Z2M shouldn't create certain bindings in another issue, could these bindings be the cause of trouble here?


<img width="1175" alt="Screen Shot 2022-01-01 at 20 10 18" src="https://user-images.githubusercontent.com/6573474/147857315-8f37386d-23ab-409d-a5bd-feb286bf3aad.png">

**Comment by SHxKM** [comment]
Well this is super weird.

I decided to try following @jeroenhoebe's advice and try once again with a  `324131092621` (first-gen remote). After several failed attempts to even start the update process, here's what happened:

1. The device started the process, got to ~35%. Then Failed.
2. The device started the process, got to ~35-40%. Then Failed. 
3. During all those attempts I held the device's hand: click off every few minutes, and read `swBuildId` every few minutes.
4. Third attempt was successful, but at the end of the process the firmware was the same version it was before the "successful" update: `5.45.1.17846`.
5. Reading `swBuildId` from the dev console also shows `5.45.1.17846`.
6. BUT, the device now shows "Check for new updates" and reports no new firmware version in the UI. Super weird because before the "successful" update, it was on the same version, and it wanted a new one. It doesn't seem to have had one installed, neither from the UI nor by reading `swBuildId`.

**Comment by Bascht74** [comment]
> 6\. BUT, the device now shows "Check for new updates" and reports no new firmware version in the UI. Super weird because before the "successful" update, it was on the same version, and it wanted a new one. It doesn't seem to have had one installed, neither from the UI nor by reading `swBuildId`.

Remove the device from Z2M, restart Z2M (e.g. restart docker / adding) and join the device again. This is probably a known bug: doesn’t show the right firmware version after OTA.

**Comment by Koenkk** [comment]
@SHxKM I don't expect binding to create any troubles

**Comment by SHxKM** [comment]
> > 6. BUT, the device now shows "Check for new updates" and reports no new firmware version in the UI. Super weird because before the "successful" update, it was on the same version, and it wanted a new one. It doesn't seem to have had one installed, neither from the UI nor by reading `swBuildId`.
> 
> Remove the device from Z2M, restart Z2M (e.g. restart docker / adding) and join the device again. This is probably a known bug: doesn’t show the right firmware version after OTA.

Please read my update. I will try this anyway.

**Comment by puijken** [comment]
Was just able to update a HUE DIMMER (324131092621) via zigbee2mqtt.
Steps followed:

1. Force remove device from z2m
2. Reset device with paper clip
3. Rejoin device to z2m
4. OTA update
5. Press button every minute
6. Wait and see updated version :)

Next steps:
- Update another dimmer switch regular way (press button once before update, click OTA update, wait)
- Restart z2m to see if first updated dimmer switch still reports correct version

Tried this multiple times before but failed every time .. only things that have changed in the meantime are newer z2m version (1.22.2) and newer coordinator firmware installed today (20211217). 


Will update here:
Current status (15h03 - dimmer updating 28%)

**Comment by SHxKM** [comment]
> Was just able to update a HUE DIMMER (324131092621) via zigbee2mqtt. Steps followed:
> 
> 1. Force remove device from z2m
> 2. Reset device with paper clip
> 3. Rejoin device to z2m (keep default name)
> 4. OTA update
> 5. Press button every minute
> 6. Wait and see updated version :)
> 
> Next steps:
> 
> * Update another dimmer switch regular way (press button once before update, click OTA update, wait)
> * Restart z2m to see if first updated dimmer switch still reports correct version
> 
> Tried this multiple times before but failed every time .. only things that have changed in the meantime are newer z2m version (1.22.2) and newer coordinator firmware installed today (20211217).
> 
> **Will update here:** _Current status second dimmer - updating regular way: 15h03 - dimmer updating 28% 15h31 - still updating, no more buttons pressed - 88% 15h43 - update done, logs showing update from version 5.45.1.17846 to 5.45.1.17846 (for the previous dimmer it showed 5.45.1.17846 to 6.1.1.28573) So update failed - will retry with procedure 1._
> 
> _Updating second dimmer - removed/rejoin device + regular button press: 15h51 - dimmer updating 3%_

Unrealistic to me because I have > 10 of those remotes and it would take days, and I would have to remap every automation, group, etc...that these entities belong to. And not that it worked for me anyway.

**Comment by puijken** [comment]
> Unrealistic to me because I have > 10 of those remotes and it would take days, and I would have to remap every automation, group, etc...that these entities belong to. And not that it worked for me anyway.

Yeah I understand, I have now done 2 and took up an entire afternoon pressing the buttons .. one more to go after this if it works. Luckily all logic/automation is in nodered, so after rejoining and updating with the default name, I can very easily rename the device again to get all up and running again.

**Comment by SHxKM** [comment]
> > Unrealistic to me because I have > 10 of those remotes and it would take days, and I would have to remap every automation, group, etc...that these entities belong to. And not that it worked for me anyway.
> 
> Yeah I understand, I have now done 2 and took up an entire afternoon pressing the buttons .. one more to go after this if it works. Luckily all logic/automation is in nodered, so after rejoining and updating with the default name, I can very easily rename the device again to get all up and running again.

Yeah using the same name is a good strategy, I can do the same. But again, too many devices in my case :)

Did method 1 work for the second remote?

**Comment by puijken** [comment]
No .. did not work .. same error again:

MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"id":"0x0017880108f1bcce","transaction":"21m3s-4"},"meta":{"device":"0x0017880108f1bcce","from":{"dateCode":"20160302","softwareBuildID":"5.45.1.17846"},"status":"update_succeeded","to":{"dateCode":"20160302","softwareBuildID":"5.45.1.17846"}},"type":"ota_update"}'

Guess I was lucky the first time.

**Comment by oscfor** [comment]
I have exactly the same problem with my 324131092621. I have updated it numerous times but only ends up with the same version. Removed, repaired, reset, new batteries, still firmware 5.45.1.17846.

**Comment by Paladin1404** [comment]
I discovered the same issue when updating my Philips HUE devices.

Hue dimmer switch with firmware 5.45.1.17846: https://www.zigbee2mqtt.io/devices/324131092621.html#philips-324131092621

Hue white ambiance Aurelle rectangle panel light with firmware 1.16.1_r19181: https://www.zigbee2mqtt.io/devices/3216331P5.html#philips-3216331p5

The update process was according to the log successful. Remove, reset and reconnect device does not help.

**Comment by SHxKM** [comment]
> I discovered the same issue when updating my Philips HUE devices.
> 
> Hue dimmer switch with firmware 5.45.1.17846: https://www.zigbee2mqtt.io/devices/324131092621.html#philips-324131092621
> 
> Hue white ambiance Aurelle rectangle panel light with firmware 1.16.1_r19181: https://www.zigbee2mqtt.io/devices/3216331P5.html#philips-3216331p5
> 
> The update process was according to the log successful. Remove, reset and reconnect device does not help.

I think your issue (at least with the Aurelle) is different and has more to do with #10131, this topic is strictly about the OTA devices (and probably only the remote at this point). I'm sure if you follow the steps I outlined in #10131 (read cluster, restart) you'll see that the Aurelle update did work, it's just that Z2M is unaware of it for some reason.

**Comment by Paladin1404** [comment]
> > I discovered the same issue when updating my Philips HUE devices.
> > Hue dimmer switch with firmware 5.45.1.17846: https://www.zigbee2mqtt.io/devices/324131092621.html#philips-324131092621
> > Hue white ambiance Aurelle rectangle panel light with firmware 1.16.1_r19181: https://www.zigbee2mqtt.io/devices/3216331P5.html#philips-3216331p5
> > The update process was according to the log successful. Remove, reset and reconnect device does not help.
> 
> I think your issue (at least with the Aurelle) is different and has more to do with #10131, this topic is strictly about the end devices (and probably only the remote at this point). I'm sure if you follow the steps I outlined in #10131 (read cluster, restart) you'll see that the Aurelle update did work, it's just that Z2M is unaware of it for some reason.

I followed the instructions from the other ticket but still the same issue for me. Both devices are not updated.

**Comment by SHxKM** [comment]
So I just tried updating a remote and have some interesting logs. I managed to keep the OTA update going until 100%, but it seems that it fails and reverts.

Here are two snippets I see when it goes near the end (the device I'm updating is called `buckram_quad_remote`):

```
Debug Got upgrade end request for '0x00XXXX0108f1d611': {"status":0,"manufacturerCode":4107,"imageType":265,"fileVersion":1107324829}
Debug Update succeeded, waiting for device to restart
Debug Device 'buckram_quad_remote' announced itself
Info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"buckram_quad_remote","ieee_address":"0x00XXXX0108f1d611"},"type":"device_announce"}'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"buckram_quad_remote"},"type":"device_announced"}'
Debug Received Zigbee message from 'buckram_quad_remote', type 'attributeReport', cluster 'genPowerCfg', data '{"batteryPercentageRemaining":200}' from endpoint 2 with groupID null
Info MQTT publish: topic 'zigbee2mqtt/buckram_quad_remote', payload '{"action":null,"action_duration":null,"battery":100,"brightness":255,"counter":1,"linkquality":183,"update":{"progress":99.53,"remaining":9,"state":"updating"},"update_available":false}'
Debug Retrieving state of 'buckram_quad_remote' after reconnect
```

and after that


```
Info Configuring 'buckram_quad_remote'
Error Publish 'set' 'read' to 'buckram_quad_remote' failed: 'Error: Read 0x00XXXX0108f1d611/1 genBasic(["swBuildId"], {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (no response received)'
Debug Error: Read 0x00XXXX0108f1d611/1 genBasic(["swBuildId"], {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (no response received) at DeconzAdapter.sendZclFrameToEndpoint (/app/node_modules/zigbee-herdsman/src/adapter/deconz/adapter/deconzAdapter.ts:641:23) at Endpoint.sendRequest (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:291:20) at Endpoint.read (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:422:28) at Object.convertSet (/app/node_modules/zigbee-herdsman-converters/converters/toZigbee.js:39:28) at Publish.onMQTTMessage (/app/lib/extension/publish.ts:246:36)
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'read' to 'buckram_quad_remote' failed: 'Error: Read 0x00XXXX0108f1d611/1 genBasic([\"swBuildId\"], {\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":true,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (no response received)'","meta":{"friendly_name":"buckram_quad_remote"},"type":"zigbee_publish_error"}'
Debug Received Zigbee message from 'buckram_quad_remote', type 'readResponse', cluster 'genPowerCfg', data '{"batteryPercentageRemaining":200}' from endpoint 2 with groupID null
Info MQTT publish: topic 'zigbee2mqtt/buckram_quad_remote', payload '{"action":null,"action_duration":null,"battery":100,"brightness":255,"counter":1,"linkquality":183,"update":{"state":"idle"},"update_available":false}'
Info Successfully configured 'buckram_quad_remote'
Error Publish 'set' 'read' to 'buckram_quad_remote' failed: 'Error: Read 0x00XXXX0108f1d611/1 genBasic(["swBuildId"], {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (no response received)'
Debug Error: Read 0x00XXXX0108f1d611/1 genBasic(["swBuildId"],
```

I don't know if this means anything.

**Comment by Tonio16** [comment]
@Koenkk  The 1.23  has still this issue. I troed to update my two 324131092621 but no success.


Antoine

**Comment by SHxKM** [comment]
@Koenkk Just tried with `1.23` and it did not succeed for me. After holding its hand with reading the `swBuildId` throughout the update, these are the logs. Goes from 0% to 100% and then just goes back to report the same verison.

```
info  2022-02-03 20:14:34: Update of 'centris_remote' at 97.67%, ≈ 1 minutes remaining
info  2022-02-03 20:14:34: MQTT publish: topic 'zigbee2mqtt/centris_remote', payload '{"action":null,"action_duration":null,"battery":100,"brightness":159,"counter":1,"linkquality":255,"update":{"progress":97.67,"remaining":60,"state":"updating"},"update_available":false}'
info  2022-02-03 20:14:34: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Update of 'centris_remote' at 97.67%, ≈ 1 minutes remaining","meta":{"device":"centris_remote","progress":97.67,"status":"update_progress"},"type":"ota_update"}'
info  2022-02-03 20:15:05: Update of 'centris_remote' at 98.86%, ≈ 0 minutes remaining
info  2022-02-03 20:15:05: MQTT publish: topic 'zigbee2mqtt/centris_remote', payload '{"action":null,"action_duration":null,"battery":100,"brightness":159,"counter":1,"linkquality":255,"update":{"progress":98.86,"remaining":30,"state":"updating"},"update_available":false}'
info  2022-02-03 20:15:05: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Update of 'centris_remote' at 98.86%, ≈ 0 minutes remaining","meta":{"device":"centris_remote","progress":98.86,"status":"update_progress"},"type":"ota_update"}'
info  2022-02-03 20:15:35: Update of 'centris_remote' at 100.00%
info  2022-02-03 20:15:35: MQTT publish: topic 'zigbee2mqtt/centris_remote', payload '{"action":null,"action_duration":null,"battery":100,"brightness":159,"counter":1,"linkquality":255,"update":{"progress":100,"remaining":30,"state":"updating"},"update_available":false}'
info  2022-02-03 20:15:35: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Update of 'centris_remote' at 100.00%","meta":{"device":"centris_remote","progress":100,"status":"update_progress"},"type":"ota_update"}'
debug 2022-02-03 20:15:37: Device 'centris_remote' announced itself
info  2022-02-03 20:15:37: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"centris_remote","ieee_address":"0x0017880108f77dd2"},"type":"device_announce"}'
info  2022-02-03 20:15:37: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"centris_remote"},"type":"device_announced"}'
info  2022-02-03 20:15:37: Finished update of 'centris_remote'
info  2022-02-03 20:15:37: MQTT publish: topic 'zigbee2mqtt/centris_remote', payload '{"action":null,"action_duration":null,"battery":100,"brightness":159,"counter":1,"linkquality":255,"update":{"state":"idle"},"update_available":false}'
debug 2022-02-03 20:15:37: Received Zigbee message from 'centris_remote', type 'readResponse', cluster 'genBasic', data '{"dateCode":"20160302","swBuildId":"5.45.1.17846"}' from endpoint 1 with groupID null
info  2022-02-03 20:15:37: Configuring 'centris_remote'
info  2022-02-03 20:15:37: Device 'centris_remote' was updated from 'null' to '{"dateCode":"20160302","softwareBuildID":"5.45.1.17846"}'
info  2022-02-03 20:15:38: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"id":"centris_remote","transaction":"6daf2-1"},"meta":{"device":"centris_remote","from":null,"status":"update_succeeded","to":{"dateCode":"20160302","softwareBuildID":"5.45.1.17846"}},"type":"ota_update"}'
info  2022-02-03 20:15:38: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/update', payload '{"data":{"from":null,"id":"centris_remote","to":{"date_code":"20160302","software_build_id":"5.45.1.17846"}},"status":"ok","transaction":"6daf2-1"}'
debug 2022-02-03 20:15:38: Received Zigbee message from 'centris_remote', type 'attributeReport', cluster 'genPowerCfg', data '{"batteryPercentageRemaining":200}' from endpoint 2 with groupID null
info  2022-02-03 20:15:38: MQTT publish: topic 'zigbee2mqtt/centris_remote', payload '{"action":null,"action_duration":null,"battery":100,"brightness":159,"counter":1,"linkquality":255,"update":{"state":"idle"},"update_available":false}'
debug 2022-02-03 20:15:39: Retrieving state of 'centris_remote' after reconnect
debug 2022-02-03 20:15:41: Received Zigbee message from 'centris_remote', type 'readResponse', cluster 'genPowerCfg', data '{"batteryPercentageRemaining":200}' from endpoint 2 with groupID null
info  2022-02-03 20:15:41: MQTT publish: topic 'zigbee2mqtt/centris_remote', payload '{"action":null,"action_duration":null,"battery":100,"brightness":159,"counter":1,"linkquality":255,"update":{"state":"idle"},"update_available":false}'
info  2022-02-03 20:15:41: Successfully configured 'centris_remote'
debug 2022-02-03 20:16:50: Received Zigbee message from 'centris_remote', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":1107314102,"imageType":265,"manufacturerCode":4107}' from endpoint 2 with groupID null
debug 2022-02-03 20:16:50: Device 'centris_remote' requested OTA
info  2022-02-03 20:16:51: MQTT publish: topic 'zigbee2mqtt/centris_remote', payload '{"action":null,"action_duration":null,"battery":100,"brightness":159,"counter":1,"linkquality":255,"update":{"state":"available"},"update_available":true}'
info  2022-02-03 20:16:51: Update available for 'centris_remote'
info  2022-02-03 20:16:51: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Update available for 'centris_remote'","meta":{"device":"centris_remote","status":"available"},"type":"ota_update"}'
debug 2022-02-03 20:17:00: Received Zigbee message from 'centris_remote', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":1107314102,"imageType":265,"manufacturerCode":4107}' from endpoint 2 with groupID null
debug 2022-02-03 20:17:00: Device 'centris_remote' requested OTA
debug 2022-02-03 20:17:10: Received Zigbee message from 'centris_remote', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":1107314102,"imageType":265,"manufacturerCode":4107}' from endpoint 2 with groupID null
debug 2022-02-03 20:17:10: Device 'centris_remote' requested OTA
debug 2022-02-03 20:17:20: Received Zigbee message from 'centris_remote', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":1107314102,"imageType":265,"manufacturerCode":4107}' from endpoint 2 with groupID null
debug 2022-02-03 20:17:20: Device 'centris_remote' requested OTA
debug 2022-02-03 20:17:30: Received Zigbee message from 'centris_remote', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":1107314102,"imageType":265,"manufacturerCode":4107}' from endpoint 2 with groupID null
debug 2022-02-03 20:17:30: Device 'centris_remote' requested OTA

**Comment by hughb8on** [comment]
Just adding this comment so I see any updates on this issue, and to let you know I have the same problem with my Hue v1 dimmers.

**Comment by domosgabor** [comment]
I have the same problem, could update only 4 of my 12 devices. Firmware version is 5.45.1.17846.

**Comment by Koenkk** [comment]
I think the hue dimmer switch requires a 2 step upgrade in some cases, but I don't have any other firmwares (https://github.com/Koenkk/zigbee2mqtt/issues/10131#issuecomment-1017841050)

**Comment by TheJulianJES** [comment]
Mhm, I also can't find any images between file version 1107314102 and 1107324829 (that update is already older than 1,5 years) for image type 265. (To me, it doesn't look like an intermediate version is required.)
Are there any other OTAs I should look for?

(Perhaps trying -- with another stick -- and ZHA's OTA process can "solve it"? Not sure what it would do do differently though, as the image seems to be successfully sent with Z2M(?))

**Comment by poonab** [comment]
Same problem here, two V1 dimmers on 5.45.1.17846 that won't update

**Comment by Koenkk** [comment]
@TheJulianJES are you able to update it with ZHA? If trying, please make a sniff when doing this: https://www.zigbee2mqtt.io/advanced/zigbee/04_sniff_zigbee_traffic.html#with-cc2531

**Comment by TheJulianJES** [comment]
@Koenkk I've already updated my two EU-style RWL021 dimmers months ago with no issues with ZHA. I don't have Hue dimmers with older firmware at the moment. (Perhaps it's possible to downgrade them though?)
I probably won't be able to provide a sniff in the near future.

**Comment by deviantintegral** [comment]
I took a look at this with my RWL020. I originally tried a normal update, which "failed" in that the firmware version is reported as the same old version, but I can also no longer apply updates as it says it's up to date.

- Firmware Build Date: 20160302
- Firmware version: 5.45.1.17846

Looking at the OTA update request:

```json
{
  "frame": {
    "Header": {
      "frameControl": {
        "frameType": 1,
        "manufacturerSpecific": false,
        "direction": 0,
        "disableDefaultResponse": false,
        "reservedBits": 0
      },
      "transactionSequenceNumber": 116,
      "manufacturerCode": null,
      "commandIdentifier": 1
    },
    "Payload": {
      "fieldControl": 0,
      "manufacturerCode": 4107,
      "imageType": 265,
      "fileVersion": 1107324829
    },
    "Command": {
      "ID": 1,
      "response": 2,
      "parameters": [
        {
          "name": "fieldControl",
          "type": 32
        },
        {
          "name": "manufacturerCode",
          "type": 33
        },
        {
          "name": "imageType",
          "type": 33
        },
        {
          "name": "fileVersion",
          "type": 35
        }
      ],
      "name": "queryNextImageRequest"
    }
  },
  "address": 33790,
  "endpoint": 2,
  "linkquality": 98,
  "groupID": 0,
  "wasBroadcast": false,
  "destinationEndpoint": 1
}
```

What confuses me is that `"fileVersion": 1107324829` matches the file version in the index:

https://github.com/Koenkk/zigbee-OTA/blob/126f99b167c2334f104099dc92a9e052fa2a9e9d/index.json#L125

```json
    {
        "fileVersion": 1107324829,
        "fileSize": 240760,
        "manufacturerCode": 4107,
        "imageType": 265,
        "sha512": "d20057d6c4e61d4ccdd3947b8a8c9bf9e126a26c080b9ce3e6839a0615b912c6f93246a60c55a44f2c8b8ec49f5041f46ac38b8a3c0a1c6ab44f062cc62089cf",
        "url": "http://fds.dc1.philips.com/firmware/ZGB_100B_0109/1107324829/Switch-ATmega_6.1.1.28573_0012.sbl-ota"
    }
```

What it seems like to me is somehow the file version got incremented, but the firmware update didn't apply? Or the firmware _is_ up to date, but reporting the wrong version?

The remote is not working to turn off three non-Philips bulbs, which is why I want to update the firmware in case it's been fixed.

https://github.com/Koenkk/zigbee2mqtt/discussions/11543

As for next steps, my thinking is to set up a local index with `force` set to `true`, to re-apply the same firmware image. Is there anything else I can do to help gather information before doing that?

**Comment by deviantintegral** [comment]
I tried the above by forcing the firmware upgrade. When I forced the upgrade, the firmware version reflected in the UI immediately updated to the latest version, and the firmware upgrade was not actually executed. This makes me think the issue is not that the OTA is failing, but that the version reporting after the OTA is complete is somehow being lost.

**Comment by zandadoum** [comment]
> I tried the above by forcing the firmware upgrade. When I forced the upgrade, the firmware version reflected in the UI immediately updated to the latest version, and the firmware upgrade was not actually executed. This makes me think the issue is not that the OTA is failing, but that the version reporting after the OTA is complete is somehow being lost.

Hello,
so, is it solved for you tho?
1) upgrade normally. goes to 100% but then reports old version
2) force upgrade, doesn't do anything but now reports correct version
is this correct? if so, because I did (1) could you show me how to do (2)?

**Comment by WatskeBart** [comment]
> I placed the following in `/opt/zigbee2mqtt/data/my_index.json`:
> 
> ```json
> [
>     {
>         "url": "Switch-ATmega_6.1.1.28573_0012.sbl-ota",
> 	"fileVersion": 1107324829,
>         "fileSize": 240760,
>         "manufacturerCode": 4107,
>         "imageType": 265,
>         "sha512": "d20057d6c4e61d4ccdd3947b8a8c9bf9e126a26c080b9ce3e6839a0615b912c6f93246a60c55a44f2c8b8ec49f5041f46ac38b8a3c0a1c6ab44f062cc62089cf",
> 	"force": true
>     }
> ]
> ```
> 
> And set the index location as shown in https://www.zigbee2mqtt.io/guide/configuration/ota-device-updates.html#ota-index-override-file
> 
> Then I triggered the upgrade in the UI.

Unfortunately it didn't work for me.
Z2M debug output was:
`ZigbeeOTA: Loading override index /config/zigbee2mqtt/my_index.json
Is new image available for '0x0017880108f1a5c7', current '{"fieldControl":0,"manufacturerCode":4107,"imageType":265,"fileVersion":1107314102}', latest meta '{"fileVersion":1107324829,"fileSize":240760,"url":"Switch-ATmega_6.1.1.28573_0012.sbl-ota","sha512":"d20057d6c4e61d4ccdd3947b8a8c9bf9e126a26c080b9ce3e6839a0615b912c6f93246a60c55a44f2c8b8ec49f5041f46ac38b8a3c0a1c6ab44f062cc62089cf","force":true}'`

But still the same version after the update finished.

**Comment by MrSpoocy** [comment]
@Koenkk, I also have the issue with my Philips LST002 (915005106701), Firmware: 1.24.0_r20239 (20170323), it say 100% but still old firmware.  I still have a Philips Hue Bridge here and I have a switch to mirror the network traffic. I use a Sonoff Zigbee 3.0 dongle on a Raspberry CM4 with Docker. If you tell me (maybe live via discord screen sharing) what exactly you need, I can make the update via Hue Bridge and maybe you see wich steps needed to work. If you want to contact me via Discord: MrSpoocy#8445.

**Comment by MrSpoocy** [comment]
@TheJulianJES http://fds.dc1.philips.com/firmware/ZGB_100B_010F/16781312/100B-010F-01001000-ConfLight-LedStrips_0012.zigbee

**Comment by TheJulianJES** [comment]
http://fds.dc1.philips.com/firmware/ZGB_100B_010F/16779778/100B-010F-01000A02-ConfLight-LedStrips_0012.zigbee should be the correct one for an intermediate update (and needs to be added to the repo with a min/max firmware version for both images)

**Comment by MrSpoocy** [comment]
Update works, I'm now at 1.50.2_r30933. But the next update not work anymore ... console say `Update of '0x00178801049b25d2' failed (Timeout: device did not request any image blocks)` after I pressed the update button :/

**Comment by deviantintegral** [comment]
I recently purchased 2 additional remotes that ship with firmware `5.45.1.17846`. I also have two bridges that are new-in-box. And, I have a conbee II I believe I can use for zigbee traffic sniffing if required. If someone could post what information we need to move this along, I can try to grab the zigbee traffic or mitm the update process from a bridge.

**Comment by puijken** [comment]
I currently have 4 dimmers with the following status:
![image](https://user-images.githubusercontent.com/77811176/178267957-7c916315-53c4-4996-b8e2-5057396d8255.png)

All have been updated or tried updating by Zigbee2MQTT only:
- 2 after multiple retries, reconnects, battery removals, etc.. are successful and running the latest version
- another one has "updated" but is consistently reporting the older version - however it seems there is no new version available when checking - also tried reconnecting, battery removal, etc..
- the last one has gone trough the process multiple times (updates, reconnects, battery removed) and keeps reporting the older version


Strange stuff ... luckily they all just work :)

**Comment by WatskeBart** [comment]
> * 2 after multiple retries, reconnects, battery removals, etc.. are successful and running the latest version

What kind of procedure would you recommend to get them updated eventually?

**Comment by hughb8on** [comment]
unstale

**Comment by umrath** [comment]
Interesting topic as I fail to update 4 Hue Dimmer devices as well, even after trying pretty much any kind of combination/clicking marathon I could come up with and retrying literally dozens of times.

**Comment by deviantintegral** [comment]
I confirmed this is still an issue with the latest release. Updating gets to 100%, but then the remote reports the old version even after removing and re-pairing.

**Comment by bjornarfjelldal** [comment]
Can confirm to be an issue for me to

**Comment by mogim0gi** [comment]
issue confirmed too

**Comment by inertia666** [comment]
Been having problems with my some of my Philips 929002398602 dimmers.

Had to force remove and take out the battery and repair to kick the start the process. Sometimes takes a few attempts to get a successful update but is ball-achingly slow.

**Comment by spicerackk** [comment]
Also having the same issue with the gen 1 dimmer switch, cannot get it to update at all, no amount of clicking or resetting changes that.

**Comment by ChrisThomas42** [comment]
Same issue here.  I've tried removing from Z2M and connecting to Hue hub and trying the update.  The dimmer does update through the Hue hub but when I connect back to Z2M the old firmware is still reported!?

Is the update actually happening just that the version is somehow misreported?

**Comment by blackie333** [comment]
The same problem with my 2 324131092621 controllers. Both are reporting version 5.45.1.17846 from 20160302.
1st was updated multiple times and stopped requesting new update. The 2nd was also updated multiple times but this one is still requesting new updates. The problem is that latest Z2MQTT versions generate notification message in HA for this new updates.
Would like to solve it for good but not sure how.

**Comment by PSTB** [comment]
> The same problem with my 2 324131092621 controllers. Both are reporting version 5.45.1.17846 from 20160302.

Exactly the same for me today. 
It has been one year with no solution. What is needed to get it fixed ?
I _can_ do the update using my Hue hub but that defeats my objective of using Z2M...

Thanks for any ideas ?!

**Comment by deviantintegral** [comment]
Well, this is fun.

I got around to setting up one of the bridges I mentioned in the summer, and ran through the update process with the official app. I have a (nearly) complete capture of a z2m update (failing), and a capture of the update from the bridge.

However:

1. The official update process seems to split sending the firmware update from actually applying it. You can see it being written to the device, and when done the app prompts to "finish" the update which it says takes 5 minutes or less. If this is part of the way all Zigbee OTAs work, and not a proprietary Philips extension, it would be a _huge_ improvement to see Z2M do this. For example, a user could configure updates to transmit overnight, and then apply and reboot the devices at a good time.
2. The app said the update was complete and showed no further updates. In the app, it showed the same version number though of something like `43.` (sorry, forgot to write it down). I repaired it back to zigbee2mqtt... and it had the old version reporting!
3. I then re-paired with the Hue bridge, and it shows a completely different version number of `1.1.28573` and no updates available.
4. I went back to z2m, and confirmed the version that shows up is `5.45.1.17846`.
5. Since I also obtained an outdoor motion sensor, and it updated firmware just fine. So while this issue says "end devices", I have at least two (another indoor motion sensor) that show that it's not _all_ indoor devices.

A brief scan of the packet captures didn't show any obvious errors or issues. The biggest flag I see seems to be in the version numbers. With Hue bulbs, the version number matches what I see in the Hue app. Perhaps we are looking at the wrong version string in the OTA files? Or Philips made a manufacturing mistake in how they manage firmware on some devices, and are hacking around it in the app?

**Comment by drifter75** [comment]
I am having the same issue, update process was successful.

Firmware build date : 20160302
Firmware version    : 5.45.1.17846

**Comment by puijken** [comment]
Have been reading and posting in this issue a couple of times in the past. Strangest results… some got updated, some did not and some reported old version but could not update anymore… I eventually gave up and left it for what it is..

Now skip a couple of months and performance /network issues have forced me to recreate my network (changed to channel 25 and all is speedy again.. wohooo). 

And the interesting part.. all my dimmers are now reporting the latest version, even the one that completely failed during updates and the one which was “stuck” on the older version… 

Maybe someone needs to try removing the device from database.db and re-pair the device and see what happen, it might give the same result.. (I actually did this but did not check versions as I was entirely focused on the performance/network stability issue).

**Comment by t34wrj** [comment]
I am also having this issue on both of my Philips 324131092621 hue dimmer switches.

If I just press 'update' in Zigbee2MQTT, the update fails with this error in the logs: 'Update of 'Philips Remote 1' failed (Device didn't respond to OTA request)'.

If I try and keep the device alive by pressing a few button after pressing the update button, the error in the logs is 'Update of 'Philips Remote 1' failed (Timeout: device did not request any image blocks').

**Comment by Luminger** [comment]
I've been struggling for a while to get a bunch of my [9290024406](https://www.zigbee2mqtt.io/devices/9290024406.html#philips-9290024406) (LWU001) updated. They won't update either with the official Hue bridge, the Bluetooth part of the official Hue Android app nor z2m.

When I got them brand new, it looked like they all requested new firmware images once (in my z2m setup). Once that happened, they're stuck on either 1.55.7_r28193 (20190322) or 1.76.10 (20201012).

When I watch for the `commandQueryNextImageRequest` dumps from z2m, I can see something strange though. All of them return `{"fieldControl":0,"fileVersion":16784918,"imageType":276,"manufacturerCode":4107}` which basically means they (believe?) they're already running the latest version, 16784918 (this should be https://github.com/Koenkk/zigbee-OTA/pull/241).

My current plan is to wait for the next round of OTAs for those bulbs and have debug logging enabled to see what's happening on the network.

Either the OTA process for those bulbs is somewhat broken or the `swBuildId` and `dateCode` attributes are unchanged in new images, which I don't think should be the case.

Alternatively another theory is that the bulbs never switched over to the new software version but have them stored in memory but were never told to boot into?

**Comment by deviantintegral** [comment]
Not stale!

**Comment by D34DC3N73R** [comment]
Add another to the list. The update completes fine, but it's "updated" to the same firmware.
```
MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/update', payload '{"data":{"from":{"date_code":"20160302","software_build_id":"5.45.1.17846"},"id":"Dimmer Switch","to":{"date_code":"20160302","software_build_id":"5.45.1.17846"}},"status":"ok","transaction":"9uxht-23"}'
```

**Comment by popy2k14** [comment]
Same issue here with a lot of Gen 1 Dimmers.

**Comment by puijken** [comment]
Did anyone try removing the device from database.db and then re-pairing?

It's been a while but I had to change my Zigbee channel, after re-pairing some devices came back with config + name (probably because they were still in database.db). On some devices however I had an issue and I think it was also the case with one of the dimmers, after removing them from database.db and re-pairing, all devices suddenly showed the newer firmware version .. 

So although I tried many times, they kept reporting an older version .. after this step suddenly all reported the latest version.

**Comment by popy2k14** [comment]
@puijken tested but didnt work. Made an update before (i think it got up to 100%). Done the steps, but firmware is still th eold one and Z2M states an uddate is available.

**Comment by popy2k14** [comment]
THanks for all the infos. I think i will live without the updates of the firmware :-)

**Comment by popy2k14** [comment]
How do you handle the update notifications?

Is it possible to disable updates for a single Z2M device?

**Comment by popy2k14** [comment]
@Koenkk i remember to my deconz/conbee II network: i had an OTA issue where you have to manually step up the firmware versions and not go from older to current with stepping over some versions.
In deconz you have to manually put the firmware files, so i stepped up once by once and that worked.
What i dont remember is which devices this issue had.

Is it possible to update to an specific version with Zigbee2MQTT to test my theory with on on the stucked dimmers?

thx

**Comment by Koenkk** [comment]
> http://fds.dc1.philips.com/firmware/ZGB_100B_010F/16779778/100B-010F-01000A02-ConfLight-LedStrips_0012.zigbee should be the correct one for an intermediate update (and needs to be added to the repo with a min/max firmware version for both images)

@TheJulianJES do you maybe also have the correct firmware for the 324131092621 (RWL020/RWL021)

**Comment by TheJulianJES** [comment]
The [image that's currently in the zigbee-ota repo](https://otau.meethue.com/storage/ZGB_100B_0109/0e1147ca-9b7e-4de1-b282-8b81c3c8e030/Switch-ATmega_6.1.1.28573_0012.sbl-ota) seems to be the only image there is currently.
It *should* work and I remember having updated RWL021s with ZHA in the past.

These images need to be sent as-is, maybe Z2M behaves somewhat differently? See: https://github.com/zigpy/zigpy/blob/1f81b162f84328eb96db18add3104d889fa6c923/zigpy/ota/image.py#L184-L274

**Comment by popy2k14** [comment]
I have exactly one "324131092621" which stucked at "5.45.1.17846". All others have 6.1.1.28573.
But i have updated them in deconz as described above. Maybe what i described above was not for 324131092621 (RWL020/RWL021).

**Comment by popy2k14** [comment]
no matter what i am trying (retry, update with/without clicking). It goes to 100% and reports old version.
Removing Device, Reseting Device (paper clip), re-pair, nothing helps...

I think i will also giv this to a friend with an hue bridge.

**Comment by popy2k14** [comment]
Paired one dimmer with an original hue bridge and it updated right away in the night without an issue.
The Bridge has an gui bug an just shows three digits, which results in: 1.1.28573.
After that i done the following:

- stopped Z2M
- removed the device from the database.json
- restart z2m and re-paired

It now shows the newest version "6.1.1.28573" and no update available.
Will do the same with the other dimmers, which are stucxk on 5.45.1.17846

**Comment by D34DC3N73R** [comment]
not stale

**Comment by indeyets** [comment]
@Koenkk is it possible to add `dont-stale` label to stop bot from coming here? the issue is very real

**Comment by popy2k14** [comment]
Not stale! Updated 3x more Dimmers on an hue bridge. Without an issue.

**Comment by SHxKM** [comment]
> Not stale! Updated 3x more Dimmers on an hue bridge. Without an issue.

Not stale indeed. Requiring a Hue Bridge is "unscalable" since some people have 20+ of those devices that they'll need to pair unpair and possibly disrupt automations for.

**Comment by popy2k14** [comment]
Same here, have 30 of those but luckily the other 26 where up-to-date. But, i had a few others in a between version to the current and the problematic one (5.45.1.17846). So maybe when youre getting once all to 6.1.1 the next update will work (when Phillips drops one). Let's see.

**Comment by nriendeau** [comment]
Hi!

I have the same problem that the firmware appears to stay stuck at the same version after an update for two Hue Dimmer Switches I have.

I say appear because as far as I can tell they did update, the problem is that Zigbee2MQTT is not aware of it and shows the previous information.

For one of my dimmers simply removing and repairing fixed it but for the other I had to remove and blocklist it, remove it from the blocklist and repair... I assume that putting it in the blocklist resulted in the device being removed from database.db, cleaned up the information in state.json and other similar files. My guess is stopping Zigbee2MQTT and editing database.db to remove the dimmer (make a backup first) and restarting Zigbee2MQTT would have given the same result.

While the UI was still showing that I was running version 5.45.1.17846 for one of the dimmers and the latest 6 (6.1.1.28573) for the other this is what I had in Zigbee2MQTT logs:

Info 2023-06-24 12:45:42MQTT publish: topic 'zigbee2mqtt/0x*************7a7', payload '{"action":"on_press","action_duration":null,"battery":100,"linkquality":180,"update":{"installed_version":1107324829,"latest_version":1107324829,"state":"idle"},"update_available":null}'

Info 2023-06-24 12:47:12MQTT publish: topic 'zigbee2mqtt/0x*************890', payload '{"action":"on_press","action_duration":null,"battery":100,"linkquality":255,"update":{"installed_version":1107324829,"latest_version":1107324829,"state":"idle"},"update_available":null}'

![hue_dimmer_switch](https://github.com/Koenkk/zigbee2mqtt/assets/505693/7a5895c2-88bf-4f8b-91ac-42d89f8655ed)

Both devices considered they were running the latest version (the same, 1107324829) with no update available but the UI was still that one of them was running an earlier one.

I am not sure how the conversion from 1107324829 to 6.1.1.28573 (external database?) is done 
but obviously 1107324829 can't be both 5.45.1.17846 and 6.1.1.28573.

Good luck and have a nice day!

Nicolas

**Comment by KoalaWerewolf** [comment]
> I placed the following in `/opt/zigbee2mqtt/data/my_index.json`:
> 
> ```json
> [
>     {
>         "url": "Switch-ATmega_6.1.1.28573_0012.sbl-ota",
> 	"fileVersion": 1107324829,
>         "fileSize": 240760,
>         "manufacturerCode": 4107,
>         "imageType": 265,
>         "sha512": "d20057d6c4e61d4ccdd3947b8a8c9bf9e126a26c080b9ce3e6839a0615b912c6f93246a60c55a44f2c8b8ec49f5041f46ac38b8a3c0a1c6ab44f062cc62089cf",
> 	"force": true
>     }
> ]
> ```
> 
> And set the index location as shown in https://www.zigbee2mqtt.io/guide/configuration/ota-device-updates.html#ota-index-override-file
> 
> Then I triggered the upgrade in the UI.

I'm in the middle of the same issue, updating from 5.45.1 to 6.1.1. I just tried the with the local index file as commented above. The update failed since I didn't have the OTA file in the same directory. However, I then updated the my_index.json file with the Philips URL (a few posts above the one I've replied to), restarted z2m, and after the restart the dimmer switch showed the correct firmware version. I never applied the update again.

I'll update another switch and see if this is repeatable.

**Comment by KoalaWerewolf** [comment]
Seems to be multiple things that can complicate thing with these switches. I had trouble getting the update to start with my next switch. After trying several thing, restart z2m, 4-button reset/restart? of the switch, I removed the battery in the switch. After the battery removal I was able to get the update to start. 

I'll post an update on the fw version when it has completed.

**Comment by KoalaWerewolf** [comment]
Repeatable with an additional twist :-)

My next dimmer switch updated and initially reported the new file version (1107324829) in its state, but the old fw in OTA. If I tried to update the fw again the fw update would actually report a fw available and then the update would run again with the same result.

The method described in issue #16148 did help me. After a new fw update completed I force removed and blocked the device. It didn't show up in the blocklist so I tried to pair it, failed because it was blocked. Unblocked the device and paired again. 

The fw was still reported as the old version, an OTA check resulted in no new fw available, but now the file version was reported as 1107324829 on the device (its state).

At this point (update successfully applied but fw reported as old version) my previous force update without an available fw-file did the trick. Local index file, try the update, the update fails since the file is not available, after a z2m restart the fw version is reported as correct.

One additional thing to note. After the failed forced update with the local index file the state of the device is that it shows Update Available in OTA which doesn't get cleared by an update attempt. I had to stop z2m, edit state.json, find my device, change "available" to "idle", save state.json, and start z2m. 

Now everything seems to be correct. The device reports the right file version, OTA reports the right fw version, and no update available.

**Comment by nriendeau** [comment]
> The method described in issue #16148 did help me. After a new fw update completed I force removed and blocked the device. It didn't show up in the blocklist so I tried to pair it, failed because it was blocked. Unblocked the device and paired again.

I think you are referring to my post in that other ticket, I provided roughly the same information in this post just before your first post a few days ago.

It was actually based on information I had found in this ticket and what I had to do to make it work for me...

As far as I can tell when we have this problem the device has already been upgraded but Zigbee2MQTT is unaware of it and still believes that the device has an earlier firmware and is unable to push the same firmware on the device a second time...

**Comment by KoalaWerewolf** [comment]
FYI my last v1 dimmer switch is continuing to misbehave. The fw update seems to be installed. The state of the device reports the right file version. FW version is reported as the old 5.45.1.

`{
    "battery": 69.5,
    "linkquality": 160,
    "update": {
        "installed_version": 1107324829,
        "latest_version": 1107324829,
        "state": "idle"
    },
    "action": "",
    "action_duration": null,
    "update_available": null
}`

But none of the workarounds works for this one. I've tried all the tricks but if I try another update it will report that an update is available, the state will shift to the installed_version being the original version and then the update can be applied again.

**Comment by nriendeau** [comment]
I would be extremely curious what another Zigbee2MQTT installation to which those remotes were never paired OR Philips own hub would say about the version of the firmware running on them.
Would it show the old version or, as I was suspecting earlier, that those remotes are already upgraded and that the information about the old version is cached somewhere...

**Comment by KoalaWerewolf** [comment]
The behavior I can see is that z2m sees `"installed_version": 1107324829` after the update, but when it queries the device the device reports the earlier version `"installed_version": 1107314102`. My guess is that the device isn't performing a final step of either applying the update or updating the fw info, resulting in this behavior. The workarounds manage to trigger the device to perform that action, but in some cases (perhaps a specific batch of devices) even the workarounds don't work.

**Comment by KoalaWerewolf** [comment]
I've paired my misbehaving dimmer with a fresh ZHA install. I moved it directly after an OTA update in z2m finished "successfully". It is reporting `5.45.1.17846` and `current_file_version : 1107314102`.

**Comment by KoalaWerewolf** [comment]
> I've paired my misbehaving dimmer with a fresh ZHA install. I moved it directly after an OTA update in z2m finished "successfully". It is reporting `5.45.1.17846` and `current_file_version : 1107314102`.

I enable OTA in ZHA, issued an update command for the dimmer switch:
```
service: zha.issue_zigbee_cluster_command
data: 
  ieee: "xx:xx:xx:xx:xx:xx:xx:xx"
  endpoint_id: 2
  cluster_id: 0x0019
  cluster_type: out
  command: 0
  command_type: client
  params:
    payload_type: 0
    query_jitter: 100
``` 

The behavior indicates that the update was at least already downloaded to the device and was applied by the update. I tried to check the progress but I couldn't find that info, so I checked the `current_file_version` and the device now reported `1107324829`.
After pairing the device with z2m again it reported the new `installed_version` and no update available. The fw version was wrong, but now the "forced update that fails" triggered the device to update its reported fw version.

Based on this I'd say that z2m is definitely doing something wrong in the way the update gets applied. 
ZHA didn't apply the update fully either, the device still reported the wrong fw version, but at least it didn't get stuck in the perpetual update available state.

**Comment by popy2k14** [comment]
@indeyets i have paired a dimmer to an hue hub after z2m did the update and stucked at the old version. Also the hue hub showed the old version. But i had updated all my dimmers without issues on the hue hub.
So yeah, i think z2m is doeing something wrong here at apply the update. The download always succeeds (as also stated by man others here).

**Comment by Gitonthegrid** [comment]
I also had an issue with updating my dimmers, but when I said "update" I was pressing the on button a couple of times, so it would connect with the Z2MQTT. After a couple of clicks, he could connect with the dimmer and started updating.

**Comment by popy2k14** [comment]
@Gitonthegrid yes, that's the normal way on battery powered devices.
Sadly there is an issue with the dimmers, hopefully just which are on version 5.45.1.17846.
See above.

Maybe yours was already on an newer version?

We will see how the update goes on our dimmers which had problematic 5.45.1.17846 before, updated with hue bridge to 6.1.1 when philips drops an new update :-)

**Comment by zen2** [comment]
I wonder if the HUE Hub doesn't use extra command for end devices:
- a command before updating to tell end device to do not go in standby mode
- a command to reset the end device after update is completed

That could explain differences of behaviour between Z2M and HUE.

**Comment by Daafip** [comment]
Pressing a button on the dimmer whilst hitting update in Z2M fixed it for me. Takes just a bit longer than when I did it throught the Hue app over bluethooth, but both took a considerable amount of time tbh.

**Comment by srd424** [comment]
Random datapoint: one of my dimmers was refusing an update, then I noticed battery was at 7%. Have swapped it for a fresh one and now it teems to be taking - fingers crossed. Maybe they check battery level and refuse if it's too low?

**Comment by popy2k14** [comment]
Sure there will be some protection against power outages during update -> a min required battery level.
But in my case i have replaced also the batteries after they failed to update on some devices.
So sadly, that did'nt solved the issue for me.

**Comment by srd424** [comment]
Yes, now I've got to the end of the update I've hit the same problems as everyone else above :( Probably going to have to put this project on the backburner now due to time constraints :(

**Comment by popy2k14** [comment]
Just pair them to a hue bridge and update will go through.

Maybe when we have the newest version installed on the dimmers, we don't have the issue again.
We will see...

**Comment by deviantintegral** [comment]
I took another look at this. I had a remote not working right and thought that I needed to update its firmware (it turned out I was just forgetting to unbind the coordinator bind when directly binding to single device, and not a group).

But, here's a complete packet capture of:

- Pairing the remote `0x00178801096545b2` / `0x817b`.
- Sending the update with the Hue bridge app
- Triggering the remote to actually apply the update
- Unpairing it

Like before, it shows the old version in zigbee2mqtt's UI.

[hue remote firmware upgrade.pcapng.gz](https://github.com/Koenkk/zigbee2mqtt/files/14967804/hue.remote.firmware.upgrade.pcapng.gz)

The network key is `a6:01:6a:70:74:39:09:25:60:88:66:d0:46:fb:84:13`. For example, in Wireshark:

<img width="846" alt="image" src="https://github.com/Koenkk/zigbee2mqtt/assets/255023/9aa3e806-d8dc-4ebb-8000-ed994a8aa6c6">

I think you can ignore `0x987e` - AFAIK, this isn't my device, and my Hue hub is unplugged 99% of the time. My best guess is a neighbour of mine somehow paired a bulb to my test network, but is normally using Bluetooth to control it.

**Comment by deviantintegral** [comment]
I was just able to get the z2m to update the device firmware data. I noticed that when I manually queried the swBuildId in the dev console, I got the correct version both in the sniffer and in z2m:

<img width="734" alt="image" src="https://github.com/Koenkk/zigbee2mqtt/assets/255023/b6003d2c-3e87-4799-9909-5fac843f082b">

I think what this means is it is likely that z2m isn't calling the basic clusters after a firmware update to get the new version information. I wonder if other types of devices send that data automatically? It is possible that the database update is where the problem is, but I think if that was the case my manual call to the cluster wouldn't have fixed things.

If we had a link to an older firmware revision (ideally `5.45.1.17846`) I could probably force a downgrade and retry. But as of now, all of my remotes are fully updated so I no longer have a reproduction case.

**Comment by paddyponchero** [comment]
Can confirm that manual method above gives correct version for me also today 👍

Had previously reconfigured, forcible removed and reset multiple times with no luck.

**Comment by popy2k14** [comment]
> I was just able to get the z2m to update the device firmware data. I noticed that when I manually queried the swBuildId in the dev console, I got the correct version both in the sniffer and in z2m:
> 
> <img alt="image" width="734" src="https://private-user-images.githubusercontent.com/255023/322223404-b6003d2c-3e87-4799-9909-5fac843f082b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNzc4MjMsIm5iZiI6MTcxNTI3NzUyMywicGF0aCI6Ii8yNTUwMjMvMzIyMjIzNDA0LWI2MDAzZDJjLTNlODctNDc5OS05OTA5LTVmYWM4NDNmMDgyYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwOVQxNzU4NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MmJiMThhMGQ0YjVkOTBiMGUwNjkwMmU2MzRjYjA1YjczMDEzOWMxNTBjODRmZmMzNzI1NmNmZGQ3NzVhNWVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.inIOjlrvo3Vx_hUfadMfhcFQABIPPSynEh4t-FTJNKg">
> I think what this means is it is likely that z2m isn't calling the basic clusters after a firmware update to get the new version information. I wonder if other types of devices send that data automatically? It is possible that the database update is where the problem is, but I think if that was the case my manual call to the cluster wouldn't have fixed things.
> 
> If we had a link to an older firmware revision (ideally `5.45.1.17846`) I could probably force a downgrade and retry. But as of now, all of my remotes are fully updated so I no longer have a reproduction case.

Nice find!! Thx a lot.
@Koenkk is it possible to read the swBuildId cluster after firmware update to solve the issue? 

Thx

**Comment by Koenkk** [comment]
Both `swBuildId` and `dateCode` are read after the update, but  if there are connection problems this might fail.

**Comment by imuncas** [comment]
Hi,

I tried it myself, and I can confirm if manually reading the swBuildId via
the Dev console, it reports the updated version:
Read result of 'genBasic': {"swBuildId":"6.1.1.28573"}
However on the device page it still reports the old version:
Firmware version5.45.1.17846
Any option to manually update in the Z2M database or force an update based
on swBuildId?

**Comment by paddyponchero** [comment]
Mine did that initially but later the page version updated.

**Comment by ChrisThomas42** [comment]
To add my observations, I have two Philips Outdoor Motion Sensors that I got a very different times.  My oldest one, let's call it A, had updated a while back but the new one, let's call it B, went through the update when I first paired it, after the update it still reported the old firmware version. A is reported in Z2M as on firmware 6.1.1.27575 (20190219), B is reported as 6.1.0.25261 (20180828).

Following what @deviantintegral suggested above, B reports in Z2M as 6.1.0.25261 but in the Dev console is reported as 6.1.1.27575.  After several hours Z2M is still reporting it as 6.1.0.2561.  Z2M also reports no updates available for device.

**Comment by ChrisThomas42** [comment]
An update.  Device B is now reporting as firmware 6.1.1.27575 but build date 20180828 rather than 20190219.

Not sure how to interpret that?

**Comment by popy2k14** [comment]
It happened again.
I bougth an old (but new) v1 dimmer switch with version 5.45.1.17846.
Updated went througth but sadly it still reports 5.45.1.17846.

Also query the swBuildId and dateCode doesn't help.

Any hint's how i can get them show the updated version without update it in an hue system?

**Comment by paddyponchero** [comment]
I came to the conclusion Philips only respond to query the swbuild once
every 12 or 24 hours. Press a button tomorrow then immeditely query swbuild

**Comment by popy2k14** [comment]
I now have done the following and it worked:

- remove device from Z2M (trash icon)
- wake device with button press during removing -> got success notification from z2m
- restarted z2m addon  (dont know if this is needed)
- waited half an hour (dont know if this is needed)
- enable join network in z2m
- press setup button 20 seconds on the switch
- device paired again and showed the correct date and build

hope this helps somebody.

**Comment by popy2k14** [comment]
@mircsicz in my case the workaround posted above worked recently.
Give it a shot.

**Comment by mircsicz** [comment]
> @mircsicz in my case the workaround posted above worked recently. Give it a shot.

For me not even the update's go through... Also have two motion sensor's and a livingcolors which I both could upgrade during the course of the WE ;-)

**Comment by popy2k14** [comment]
> > @mircsicz in my case the workaround posted above worked recently. Give it a shot.
> 
> For me not even the update's go through... Also have two motion sensor's and a livingcolors which I both could upgrade during the course of the WE ;-)

Ok, that's weird.

In my experience (and i think others in here) is, that the update went through but the device version is not showing the correct version in Z2M.

I think your issue is another one.
Some things you could try:

- Re-pair it directly to the coordinator (near the coordinator). Use the method above from my workaround (remove device first with waking it and than repair it)
- Keep the device alive during update -> press a button once every few seconds
- maybe (but rather not) replacing the battery helps?

Good luck

**Comment by mircsicz** [comment]
![IMG_0089](https://github.com/Koenkk/zigbee2mqtt/assets/7398305/21d5362b-bcff-4307-8902-ad29bb134c5f)
I'm sitting right next to it...

I've repaired it a few times during the course of the WE and also tried updating with clicking/tapping I Button last night and it didn't go through. And also the 2450s are brand new ;-)

**Comment by mircsicz** [comment]
I still have the old bridge around...

I'll report back 😉

**Comment by Kodemikkel** [comment]
I had the same issue with a Philips hue switch (324131092621) on firmware version 5.45.1.17846 - it would not accept any OTA updates, logging "Did not respond to OTA request".

The reported battery state in Z2M was 26.5% (although I never really trust the reported battery state), so I went about changing the battery (the new battery reported 90% in Z2M) and the switch then accepted the update. As of writing, it is still updating, but that's a step further than before I changed the battery.

Just posting this so it might help others with the same issue.

**Comment by jonnycastaway** [comment]
@Kodemikkel The biggest question is: did the update come to an regular end? With reporting of new Firmware in z2m?
I have a dimmer switch that updates tens of times. It but it never reported the new firmware to z2m. I rejoined, reset, put out and in the battery but nothing.
3 other dimmer switches have done this update without problems.

**Comment by popy2k14** [comment]
@Kodemikkel @jonnycastaway that's the real question :-)

**Comment by Kodemikkel** [comment]
@jonnycastaway, @popy2k14. I was AFK when the update finished the first time, but it didn't go through.
I tried updating once more, and it came to 100% without any error messages, but unfortunately the update still didn't go through.
So I guess that didn't work after all.

**Comment by popy2k14** [comment]
@Kodemikkel yeah, i thought i would end in this state. I think only thing to get these updated for me was to pair it to an hue hub, update it and repair it with Z2M.

**Comment by Kodemikkel** [comment]
> @Kodemikkel yeah, i thought i would end in this state. I think only thing to get these updated for me was to pair it to an hue hub, update it and repair it with Z2M.

I actually have another one of these switches, and that one has upgraded perfectly fine to 6.1.1.28573, but the particular switch I mentioned above does not want to upgrade for some reason

**Comment by SHxKM** [comment]
More than 3 years later...

Is there any way to sniff/discover whatever it is that the Philips bridge does "right" and Z2M "doesn't"? 

There's gotta be a better way than using the Hue bridge..

**Comment by mundschenk-at** [comment]
I've just seen this issue for the first time, all my 2021 dimmer switches and SML03 updated fine, but the original dimmer switches stayed on the same firmware after getting to 100 %. What I did notice during the update process is that the dimmer seemed to constantly rejoin the network:

```
z2m: Accepting joining not in blocklist device '0x0123456789abcdef'
```

It triggered the update a second time, closer to the coordinator. So far, no re-joins, maybe this time the update will take? Could the difference be whether the device connects directly to the coordinator or via a router?

**Comment by Gunni** [comment]
I managed to update mine by setting reporting intervals to 1-5 seconds, so that it's basically always awake. After the upgrade they reset themselves but they upgraded fairly fast.

**Comment by danir-de** [comment]
> I managed to update mine by setting reporting intervals to 1-5 seconds, so that it's basically always awake. After the upgrade they reset themselves but they upgraded fairly fast.

How exactly did you set up the reporting settings in Z2M for it to work? Can't seem to replicate it.

**Comment by Struisvogel** [comment]
Seeing this is still open ...

Had this on 1 RWL022 dimmer, used on zigbee2mqtt 2.1.1-1 in Home Assistant (HA). The button pressing to keep alive wasn't doing it for me. Nothing in z2m is set different from default in HA.

What got the update started, to completion and successful reporting was:
- Start the update. The log indicates the update is started but of course nothing happens. Then quickly go...
- Reconfigure the dimmer. This shows reconfiguring, and once that's done it jumped into the update. Not clear to me what this does.

from there the update completed in 20 minutes and reported the correct version.

[RWL022 FW Update.txt](https://github.com/user-attachments/files/18701204/RWL022.FW.Update.txt)

**Comment by mundschenk-at** [comment]
> I managed to update mine by setting reporting intervals to 1-5 seconds, so that it's basically always awake. After the upgrade they reset themselves but they upgraded fairly fast.

Interesting, I managed to update one of mine with that method, but I needed to re-interview to make Z2M recognize the updated version.

**Comment by mundschenk-at** [comment]
> The device that reports the firmware correctly had no bindings to a bulb set, the other one had a binding to a bulb. So what are others experiencing or is the one experience just a coincidence or is there something here to investigate further?

None of my v1 dimmers have direct bindings, but do have the issue with updating, so I think this is probably coincidental.

**Comment by KennethLavrsen** [comment]
I have just been updating around 15 of the 4 button Hue dimmers. It went well until the last two. They both OTA update and all seems normal and finishes. But they both report the old version and are marked as needing and upgrade. I tried to unpair and repair. No luck. Once Zigbee2mqtt has gone angry at a dimmer it just will not succeed.  Any other work arounds to try?

**Comment by ChrisThomas42** [comment]
> None of my v1 dimmers have direct bindings, but do have the issue with updating, so I think this is probably coincidental.

That's what I was thinking was more likely, but I thought I'd mention it on the off chance.

**Comment by KennethLavrsen** [comment]
I have tried to OTA update one device 4 times now and the other 3 times. All looks normal. The upgrade runs for 45 minutes. And then the device ends up with the original version anyway. I have tried to manually read the versions. I have clicked interview followed by reconfigure. And always while clicking the remote so it is alive. Nothing works. These two out of 15 just wont upgrade. And I cannot see if it is because Z2M will not read version, or the upgrade actually fails. I unpaired and repaired. One would think Z2M would forget the data when you do that. I wonder if the OTA actually uploads the wrong old file??

**Comment by mundschenk-at** [comment]
No, if it took the update, OTA will not let you update a second time, even when Z2M still shows the old version. If the update shows up as available in the OTA tab, then the dimmer didn't apply it.

**Comment by boadk** [comment]
I could not start Update, I changed batteries in my hue motion sensor SML001 (battery went from 1 to 100%) and started update and activated motion in front of the sensor, and now it started updating, I will report results when finished.

**Comment by kaiwoknaj** [comment]
I also hat issues updating, then I deleted my [Philips 324131092621](https://www.zigbee2mqtt.io/devices/324131092621.html#philips-324131092621) from Z2MQTT, used a new battery, paired, updated.
WORKS

**Comment by kimheino** [comment]
> Put MQTT logging in info so you see the popups First reset the Philips Hue remote by holding all 4 buttons down for 10 seconds, Wait till it reconnects. Click the on button to speed that up. Then interview device (click the blue "i" icon in the device about screen in MQTT and verify that the interview succeeds Then click reconfigure Now the version is right

Thank you. This worked for me too.

**Comment by nriendeau** [comment]
> And Z2M has a problem showing the new version after the interview and also needs to reconfigure

This is more or less the conclusion some of us came up with about two years ago...

re: https://github.com/Koenkk/zigbee2mqtt/issues/10139#issuecomment-1605762725

**Comment by honzis1364** [comment]
Me three. Every HUE Accessory will result in fail in OTA. Either wont't start the OTA (timeout). Or the update will start, but after some period of a time will fail by Restarting the adapter.
I have a solid 150 devices network, good connection, no interference, everything is rock solid - except the OTA of HUE Dimmer Switches (both Gen 1 and 2) and Motion sensors.
I've tried it on EZSP 6.x.x as well as on Amber 7.4.4 versions of the SONOFF E coordinator. I've tried setting lower package size in OTA, pressing the buttons to wake accessories up, setting ping mode to accessories every 5 secs, nothing has worked for me.

**Comment by ChrisThomas42** [comment]
@KennethLavrsen Your find of reset, interview, update has worked on all remaining 10 dimmer switches, well done and thank you!  ALL of my dimmer switches are now reporting the latest firmware version.

All the devices I updated using this method were on the old firmware and reporting update available.  On the last couple I updated I tried just resetting and then updating but this didn't work, I needed to reset, interview and then update.

3 of the devices changed the bindings I had to a light bulb, but once I noticed this I just changed the binding.

I've had one stubborn switch that took a couple of tries to take the update but it worked eventually.  I just left about a minute after the interview completed and that seemed to do the trick.  I also had to keep it awake by pressing a button every few minutes, plus it kept giving "... Accepting joining..." messages during the update.

Also this interview, reconfigure method worked for my motion sensors as well.

Thank you Kenneth!

**Comment by imuncas** [comment]
Now this is super strange. Mine looks like this:
[image: Screenshot 2025-02-10 at 19.12.22.png]
I even scanned through all settings if you can disable it somewhere. I
found a thread that you can trigger from home assistant. I'll try that
one...

**Comment by imuncas** [comment]
The missing button was not about resolution, all the other butons were visible. (The buttons are also in the device About page, all the way in the bottom) I upgraded to 2.x and now I have the inteview buttons. 
I'm still not able to upgrade some of the dimmers, 7 updated properlu, 5 is still stuck on version 5.45.1.17846
During the interview I get the following info message, not sure if related:
info 2025-02-15 17:10:00zh:controller:device: Device '0x00178801086b6073' is only compliant to revision 'pre-21' of the ZigBee specification (current revision: 23).
I don;t think the reset works well though. No matter which dimmer I use, pressing the 4 buttons no matter how long - I tried holding for more then a minute even - only lights up the green led after 3 seconds, and then it say lit. On release it blinks breiefly in multiple colors, and then it's already usable in 2-3 seconds.
If I use the setup button for 10s, I have to permit joining of devices before I do the interview.
No matter which part I use, 4 buttons or setup, after the interview and configuration it still on the old version. :(

**Comment by makanimike** [comment]
I had difficulties updating my Hue remote as well. It required 2-3 attempts, and then would always failed.  
In the end, I got it to work by occasionally pressing a button during the update process. I thought, maybe the remote is going to sleep during the update, because it takes a good 45 minutes or longer. So, I just pressed a button every 15 minutes or so. That worked on the first try.

**Comment by sybux** [comment]
Hi,
also trying to update Philipps Dimmer to lastest version. Update is going on. Took time. Finish as 100%. but 1 day later, update is available again as if the device wasn't updated.

I've tested with a fresh new battery but same problem of update.

**Comment by babyhuehnchen** [comment]
> I also hat issues updating, then I deleted my [Philips 324131092621](https://www.zigbee2mqtt.io/devices/324131092621.html#philips-324131092621) from Z2MQTT, used a new battery, paired, updated. WORKS

That might actually be the solution in my case as my batteries are in the single digits, ill give that a try

**Comment by nriendeau** [comment]
About two years ago I said that removing and re-adding had fixed things for me (see https://github.com/Koenkk/zigbee2mqtt/issues/10139#issuecomment-1605762725) but what fixed things for me this time was updating then simply re-interviewing by clicking the little icon with an "i", after that the version displayed was the new version.

**Comment by PartyingChair** [comment]
Having same issue with https://www.zigbee2mqtt.io/devices/324131092621.html#philips-324131092621 and no matter how many times I try to update it, it keeps failing. I had a couple say they were complete, then later say they weren't actually complete.

**Comment by OneCreek** [comment]
> Having same issue with https://www.zigbee2mqtt.io/devices/324131092621.html#philips-324131092621 and no matter how many times I try to update it, it keeps failing. I had a couple say they were complete, then later say they weren't actually complete.

I have the same Issue with the following Firmware:
![Image](https://github.com/user-attachments/assets/9488d7cb-9777-44e3-a126-46dca05764a9)

What I get with the device following https://github.com/Koenkk/zigbee2mqtt/issues/10139#issuecomment-2749525677 is the response: 
`Device is only compliant to revision 'pre-21' of the ZigBee specification (current revision: 23).`
Not sure if this is a relevant information for the issue though....

**Comment by OneCreek** [comment]
> I have the same Issue with the following Firmware: 

Update: ["repairing the device should fix it"](https://github.com/Koenkk/zigbee2mqtt/issues/5869#issuecomment-785347503) from #5869 helped solving my issue

**Comment by puijken** [comment]
Since the latest updates (currently running 2.6.3) with firmware ZStack3x0 (20230507) and I have been able to update both old/new style Hue dimmers recently.

For it to start I had to pull the battery first for some of the dimmers .. but after that .. push a button a couple of times, click update and they seem to happily update themselves. Reporting correct firmware versions after.

Cool!

**Comment by sybux** [comment]
For mine, I've able to update them (6.1.1.28573 (20190410) | 4.2.00-6.f.9d | 4.3.00-7.3.05) but needed to do it 3-4 times.

Process was to :
1. remove battery
2. click the "update device firmware" BUT NOT OK
3. insert battery and in the same second click the OK button for update.

For better result, a new battery is welcome.
Update will succeed for 1 day about, revert to previous version. Redo the process 2 or 3 times and at the end update is done.

**Comment by AlexHunterCodes** [comment]
After a lot of trial and error trying to update a specific gen 1 dimmer switch with no success (the others worked on fresh batteries), I dug out my Hue Bridge v2 again and it updated the remote first time. The remotes don't appear to ever request update blocks from the coordinator on Z2M, so Philips must be doing something special to account for firmware bugs or limitations in those early devices.

**Comment by ChrisThomas42** [comment]
ARGGGHHH! All of my dimmers have had a firmware update, about half successfully completed the update without intervention.  The rest required the 4 button reset, interview and reconfigure before they successfully updated.

The BIG problem for me is that some of my switches have bindings to a nearby lamp, after the update NONE of the bindings work, for ANY of the dimmers that have bindings I created.  Worse is that this morning I've deleted and recreated the binding on two of the dimmers and the binding still do not work.  On one dimmer I have deleted the dimmer from Z2M, reset the dimmer (via the button), added the dimmer to Z2M and added the binding BUT the binding still does not work.

I'm not sure what to do now as I've now updated all my dimmers, more tinkering required...
